### ISSUE TITLE
Email formatting changed in Python 3.8

### ISSUE THREAD

**Issue by alex-ber** [alex-ber/AlexBerUtils#7]
I have found regression in email.message.EmailMessage. This changes how e-mail is looks like when generated using EmailMessage.

I’m creating it as following:

```python3
from email.message import EmailMessage as _EmailMessage
from email.policy import SMTPUTF8 as _SMTPUTF8

msg = _EmailMessage(policy=_SMTPUTF8)

msg['Subject'] = self.get_subject()
msg.set_content(body.getvalue(), subtype='html')
```

Note: bodyText has lines that are longer than 78 not counting the end of line character(s).

> Quote: max_line_length
> The maximum length of any line in the serialized output, not counting the end of line character(s). Default is 78, per [RFC 5322](https://tools.ietf.org/html/rfc5322.html). A value of 0 or None indicates that no line wrapping should be done at all.

[https://docs.python.org/3/library/email.policy.html#module-email.policy](https://docs.python.org/3/library/email.policy.html#module-email.policy)

At Python 3.7 this code generate msg with header that has

_('Content-Type', 'text/html; charset="utf-8"'),_ **('Content-Transfer-Encoding', '7bit')**

At Python 3.8 this same code generate msg with header that has

_('Content-Type', 'text/html; charset="utf-8"')_, **('Content-Transfer-Encoding', 'quoted-printable')** 

**This effectively breaks the line apart.**

***********************************************************************************************************************************

I’m fixing the code to:

```python3
from email.message import EmailMessage as _EmailMessage
from email.policy import SMTPUTF8 as _SMTPUTF8

msg = _EmailMessage(policy=_SMTPUTF8)

msg['Subject'] = self.get_subject()
msg.set_content(body.getvalue(), subtype='html', cte='8bit')
```

I’m using 8bit and not 7bit because I want to support Unicode character (in my example, there is no surrogate pair, so it was resolved to 7bit). See the code below, it will be more clear.

**********************************************************************************************************************************
The cause of the change is in the file `email.contentmanager`

This is the Python 3.7 function:

```python3
def _encode_text(string, charset, cte, policy):
    lines = string.encode(charset).splitlines()
    linesep = policy.linesep.encode('ascii')
    def embedded_body(lines): return linesep.join(lines) + linesep
    def normal_body(lines): return b'\n'.join(lines) + b'\n'
    if cte==None:
        # Use heuristics to decide on the "best" encoding.
        try:
            return '7bit', normal_body(lines).decode('ascii')
        except UnicodeDecodeError:
            pass
        if (policy.cte_type == '8bit' and
                max(len(x) for x in lines) <= policy.max_line_length):
            return '8bit', normal_body(lines).decode('ascii', 'surrogateescape')
        sniff = embedded_body(lines[:10])
        sniff_qp = quoprimime.body_encode(sniff.decode('latin-1'),
                                          policy.max_line_length)
        sniff_base64 = binascii.b2a_base64(sniff)
        # This is a little unfair to qp; it includes lineseps, base64 doesn't.
        if len(sniff_qp) > len(sniff_base64):
            cte = 'base64'
        else:
            cte = 'quoted-printable'
            if len(lines) <= 10:
                return cte, sniff_qp
    if cte == '7bit':
        data = normal_body(lines).decode('ascii')
    elif cte == '8bit':
        data = normal_body(lines).decode('ascii', 'surrogateescape')
    elif cte == 'quoted-printable':
        data = quoprimime.body_encode(normal_body(lines).decode('latin-1'),
                                      policy.max_line_length)
    elif cte == 'base64':
        data = _encode_base64(embedded_body(lines), policy.max_line_length)
    else:
        raise ValueError("Unknown content transfer encoding {}".format(cte))
    return cte, data
```
Note the lines

```python3
    if cte==None:
        # Use heuristics to decide on the "best" encoding.
```

This is Python 3.8 function:

```python3
def _encode_text(string, charset, cte, policy):
    lines = string.encode(charset).splitlines()
    linesep = policy.linesep.encode('ascii')
    def embedded_body(lines): return linesep.join(lines) + linesep
    def normal_body(lines): return b'\n'.join(lines) + b'\n'
    if cte==None:
        # Use heuristics to decide on the "best" encoding.
        if max((len(x) for x in lines), default=0) <= policy.max_line_length:
            try:
                return '7bit', normal_body(lines).decode('ascii')
            except UnicodeDecodeError:
                pass
            if policy.cte_type == '8bit':
                return '8bit', normal_body(lines).decode('ascii', 'surrogateescape')
        sniff = embedded_body(lines[:10])
        sniff_qp = quoprimime.body_encode(sniff.decode('latin-1'),
                                          policy.max_line_length)
        sniff_base64 = binascii.b2a_base64(sniff)
        # This is a little unfair to qp; it includes lineseps, base64 doesn't.
        if len(sniff_qp) > len(sniff_base64):
            cte = 'base64'
        else:
            cte = 'quoted-printable'
            if len(lines) <= 10:
                return cte, sniff_qp
    if cte == '7bit':
        data = normal_body(lines).decode('ascii')
    elif cte == '8bit':
        data = normal_body(lines).decode('ascii', 'surrogateescape')
    elif cte == 'quoted-printable':
        data = quoprimime.body_encode(normal_body(lines).decode('latin-1'),
                                      policy.max_line_length)
    elif cte == 'base64':
        data = _encode_base64(embedded_body(lines), policy.max_line_length)
    else:
        raise ValueError("Unknown content transfer encoding {}".format(cte))
    return cte, data
```

note, the lines:

```python3
    if cte==None:
        # Use heuristics to decide on the "best" encoding.
        if max((len(x) for x in lines), default=0) <= policy.max_line_length:
```

The line in bold was added in Python 3.8. 
Because I do have long lines the try block that tries 7bit and 8bit is skipped in Python 3.8 
And it is resolved as quoted-printable.

**Comment by alex-ber** [comment]
Note, that it is possible that 7bit will be replaced with 8bit, that I'm considering as ok.
This fix will be released soon.

**Comment by alex-ber** [comment]
Released in 0.6.0a4
